.github/workflows/e2e-detox-android.yml

```diff
@@ -0,0 +1,109 @@
+# by using 'E2E iOS tests for PR (LBW 1)' instead.
+name: Detox E2E Android Tests PR
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+
+  build-android-apk:
+    runs-on: macos-14
+    env:
+      ORG_GRADLE_PROJECT_jvmargs: -Xmx8g
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: ci/prepare-android-build
+        uses: ./.github/actions/prepare-android-build
+        env:
+          STORE_FILE: "${{ secrets.MM_MOBILE_STORE_FILE }}"
+          STORE_ALIAS: "${{ secrets.MM_MOBILE_STORE_ALIAS }}"
+          STORE_PASSWORD: "${{ secrets.MM_MOBILE_STORE_PASSWORD }}"
+          MATTERMOST_BUILD_GH_TOKEN: "${{ secrets.MATTERMOST_BUILD_GH_TOKEN }}"
+
+      - name: Cache Gradle dependencies
+        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Install Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: "17"
+          distribution: "adopt"
+          cache: "gradle"
+
+      - name: Validate Gradle wrapper
+        uses: gradle/actions/wrapper-validation@v3
+
+      - name: Inject Detox settings
+        run: cd detox && npm run e2e:android-inject-settings
+
+      - name: Update minSdkVersion for react-native-image-picker
+        run: |
+          sed -i '' 's/minSdkVersion 21/minSdkVersion 23/' ./node_modules/react-native-image-picker/android/build.gradle
+          cat ./node_modules/react-native-image-picker/android/build.gradle | grep minSdkVersion
+
+      - name: Detox build
+        run: |
+          cd detox
+          npm install
+          npm install -g detox-cli
+          npm run e2e:android-build
+
+      - name: ci/upload-android-pr-build
+        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
+        with:
+          name: android-build-apk-${{ github.run_id }}
+          path: "android/app/build/outputs/apk/**/app-*.apk"
+
+  e2e-android-on-mac:
+    name: android-detox-e2e
+    continue-on-error: true
+    runs-on: macos-13
+    timeout-minutes: 40
+    needs:
+      - build-android-apk
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+
+    - name: ci/prepare-node-deps
+      uses: ./.github/actions/prepare-node-deps
+
+    - name: Install Sharp CLI for faster image generation during prebuild
+      run: npm install --global sharp-cli
+
+    - name: Start React Native Metro Server
+      run: npm run start &
+
+    - name: Install Detox Dependencies
+      run: cd detox && npm i
+
+    - name: Download iOS Simulator Build
```

Comment:
Probably should be renamed to `Download Android apk build`?

Reply:
Yes,
I'll remove this workflow and follow up with a Build-Test workflow.